src/utils/buffered.ts

```diff
@@ -0,0 +1,55 @@
+import { Transform } from 'stream';
+
+const MIN_SIZE = 8 * 1024;
+
+export default class Buffered extends Transform {
+  private chunks: Buffer[] | null;
+  private timer: number | null;
+  constructor() {
+    super({
+      readableHighWaterMark: 10 * 1024 * 1024,
+      writableHighWaterMark: 10 * 1024 * 1024,
+    } as any);
+    this.chunks = null;
+    this.timer = null
+  }
+
+  sendData() {
+    let { chunks } = this;
+    if (chunks) {
+      this.chunks = null;
+      let buf = Buffer.concat(chunks);
+      this.push(buf);
+    }
+  }
+
+  _transform(chunk: Buffer, encoding: any, callback: any) {
+    let { chunks, timer } = this;
+    if (timer) clearTimeout(timer)
+    if (chunk.length < MIN_SIZE) {
+      if (!chunks) return callback(null, chunk);
+      chunks.push(chunk);
+      this.sendData();
+      callback();
+      return;
+    }
+    if (!chunks) {
+      chunks = this.chunks = [chunk];
+    } else {
+      chunks.push(chunk);
+    }
+    this.timer = setTimeout(this.sendData.bind(this), 100);
```

Comment:
Do we need the 100ms timeout? Would it work if we just delayed `sendData` to the next execution cycle (e.g. with 1ms timeout)?

There's an edgecase if chunk size is exactly a multiple of `8*1024`, then `sendData` will have an added 100ms delay to it.

Reply:
I'm tried with 1ms and 10ms timeout, it doesn't work most of times, changed to 20ms instead.

Reply:
OK, we may want to change the buffering strategy or adopt a buffered stream dependency, but I think this is a big improvement and worthwhile compromise. Thanks!